pubnub-gson/src/main/java/com/pubnub/internal/endpoints/presence/SetStateImpl.java

```diff
@@ -0,0 +1,74 @@
+package com.pubnub.internal.endpoints.presence;
+
+import com.pubnub.api.PubNubException;
+import com.pubnub.api.builder.PubNubErrorBuilder;
+import com.pubnub.api.endpoints.presence.SetState;
+import com.pubnub.api.endpoints.remoteaction.ExtendedRemoteAction;
+import com.pubnub.api.endpoints.remoteaction.MappingRemoteAction;
+import com.pubnub.api.models.consumer.presence.PNSetStateResult;
+import com.pubnub.internal.PubNubCore;
+import com.pubnub.internal.endpoints.DelegatingEndpoint;
+import lombok.Setter;
+import lombok.experimental.Accessors;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+@Setter
+@Accessors(chain = true, fluent = true)
+public class SetStateImpl extends DelegatingEndpoint<PNSetStateResult> implements SetState {
+    private List<String> channels = new ArrayList<>();
+    private List<String> channelGroups = new ArrayList<>();
+    private Object state;
+    private String uuid = pubnub.getConfiguration().getUuid();
+    private boolean withHeartbeat;
+
+
+    public SetStateImpl(PubNubCore pubnub) {
+        super(pubnub);
+    }
+
+    @Override
+    protected void validateParams() throws PubNubException {
+        if (state == null) {
+            throw new PubNubException(PubNubErrorBuilder.PNERROBJ_STATE_MISSING);
+        }
+
+        String stringifiedState = pubnub.getMapper().toJson(state);
```

Comment:
I'm guessing that's because `channels` is initialized to be non-null (`channels = new ArrayList<>();`) and so the user can't forget to set it, as opposed to arguments that start off as `null` and the user HAS to set them before calling sync/async